src/controllers/carpool.ctrl.ts

```diff
@@ -88,6 +88,49 @@ module CarpoolController {
         next(new restify.InternalServerError(err.message))
       });
   }
+
+  export function updateCarpool(req:restify.Request, res:restify.Response, next:restify.Next) {
+    var carpoolID = req.params.carpoolID;
+
+    function getIDFromHref(href:string) : string {
+      var hrefComponents = href.split('/');
+      return hrefComponents[hrefComponents.length - 1];
+    }
+
+    function addFieldFromJSON(JSONObject, newObject, fieldName:string) {
+      if (JSONObject[fieldName]) {
+        newObject[fieldName] = JSONObject[fieldName];
+      }
+      return newObject;
+    }
+
+    Promise.resolve()
+      .then(() => {
+        return addFieldFromJSON(req.body, {}, "name");
+      })
+      .then((obj) => {
+        return addFieldFromJSON(req.body, obj, "description");
+      })
+      .then((obj) => {
+        if (req.body.campus) {
+          obj["campus"] = getIDFromHref(req.body.campus);
+        }
+        return obj;
+      })
+      .then((carpoolUpdate) => {
+        return carpoolService.updateCarpool(carpoolID, carpoolUpdate);
+      })
+      .then(() => {
+        res.send(204);
+      })
+      .catch(errors.CarpoolNotFoundException, (err) => {
+        next(new restify.NotFoundError(err.message));
+      })
+      .catch((err) => {
+        next(new restify.InternalServerError(err.message));
+      })
+      .then(next);
+  }
```

Comment:
Nice. This refactor also helped fix another issue i just discovered with the carpool not found exception not showing up